sdks/python/apache_beam/typehints/trivial_inference.py

```diff
@@ -227,6 +227,9 @@ def element_type(hint):
     return hint.inner_type
   elif isinstance(hint, typehints.TupleHint.TupleConstraint):
     return typehints.Union[hint.tuple_types]
+  elif isinstance(hint,
+                  typehints.UnionHint.UnionConstraint) and not hint.union_types:
```

Comment:
Can you help me better understand why we're also requiring the `not hint.union_types` part?

Reply:
The empty union is a type that can never happen; we want to propagate that to its (non-existent) "element" type.